lib/Runtime/Library/JavascriptArray.inl

```diff
@@ -955,45 +955,45 @@ SECOND_PASS:
             return true;
         }
 
-        SparseArraySegment<T> *toSegment = PrepareSegmentForMemOp<T>(toStartIndex, length);
-        if (toSegment == nullptr)
+        const auto isSegmentValid = [length](Js::SparseArraySegment<T>* segment, uint32 startIndex) {
+            uint32 end, segmentEnd;
+            // Check the segment is long enough
+            return (
+                segment &&
+                !UInt32Math::Add(startIndex, length, &end) &&
+                !UInt32Math::Add(segment->left, segment->length, &segmentEnd) &&
```

Comment:
Hmm.. if PrepareSegmentFromMemOp does the right thing, then this part should be fine.

---
In reply to: [59632827](https://github.com/Microsoft/ChakraCore/pull/793#discussion_r59632827) [](ancestors = 59632827)